src/EntityFramework.Storage/src/Stores/DeviceFlowStore.cs

```diff
@@ -60,43 +59,41 @@ public class DeviceFlowStore : IDeviceFlowStore
         /// <param name="userCode">The user code.</param>
         /// <param name="data">The data.</param>
         /// <returns></returns>
-        public virtual Task StoreDeviceAuthorizationAsync(string deviceCode, string userCode, DeviceCode data)
+        public virtual async Task StoreDeviceAuthorizationAsync(string deviceCode, string userCode, DeviceCode data)
         {
             Context.DeviceFlowCodes.Add(ToEntity(data, deviceCode, userCode));
 
-            Context.SaveChanges();
-
-            return Task.FromResult(0);
+            await Context.SaveChangesAsync();
```

Comment:
Coulnd't this be returned directly, without compiling a state machine?

Reply:
you mean something like:
`return context.SaveChangeAsync()` ?

Reply:
Yes and remove the `async` keyword 😄

Reply:
good suggestion, I didn't know that would have performance impact
I will look again on the places where this could be implemented

Reply:
Here's an example of a compilation with and without `async`/`await`; https://gist.github.com/khellang/6e7b3bd197657112714c.

Reply:
Doing this hides where the exception happens.. don't remove async / await unless it's extremely performance critical. And since this has a DB call, it isn't really.

Reply:
@jepperc point made sense to me, reverted the changes